### PR TITLE
Nav-Unification: Shows applicable menu items when no domain is provided

### LINKED ISSUE #48824: Nav Unification: All Sites mode
All Sites mode is currently broken as Nav Unification seems to expect a selected site at all times.

#### Steps to reproduce the behavior

1. Go to https://wordpress.com/posts
2. Click on Upgrades > Domains (should work, but without a null query arg)
3. Click on Upgrades (site selector shows up, not an All Sites screen)

image

### PR DESCRIPTION
**Pull request by cpapazoglou**
#### Changes proposed in this Pull Request

* Adds an array with menu items supporting all sites screen
* Shows the above menu items when no Domain is provided

#### Testing instructions

* `/posts`
* Click on Upgrades > Domains (should work, without a null query argument)

Before | After
-------|------
 | 

Fixes #48824

### CODE REVIEW

File: client/my-sites/sidebar-unified/use-site-menu-items.js [hunk 1]
```diff
@@ -46,6 +47,13 @@ const useSiteMenuItems = () => {
 		canCurrentUser( state, selectedSiteId, 'edit_theme_options' )
 	);
 
+	/**
+	 * When no site domain is provided, lets show only menu items that support all sites screens.
+	 */
+	if ( ! siteDomain ) {
```
Comment (getdave) [inline client/my-sites/sidebar-unified/use-site-menu-items.js:53]: Should we also bale out of the `useEffect`s above when there is no site domain as these calls will not be required.

**Review by getdave — APPROVED** [review]
I tested this and there didn't seem to be any problems once in All Sites view everything loaded and worked as expected including `Domains`.

One minor nit noted otherwise all good!

I will say that I think we should considering organising all the static data files under a single directory to keep them organised but that can be handled in a follow up.

Nice work 👏 

Screen Shot 2021-01-19 at 14 54 12

File: client/my-sites/sidebar-unified/use-site-menu-items.js (hunk 1 above)
Comment (cpapazoglou) [reply client/my-sites/sidebar-unified/use-site-menu-items.js:53]: I had already tested that we don't call the api with a `null` Domain and it seems it is safeguarded. But you are right! I have addressed it here https://github.com/Automattic/wp-calypso/pull/49026/files together with the "organising all the static data files".

File: client/my-sites/sidebar-unified/use-site-menu-items.js [hunk 2]
```diff
@@ -24,11 +25,11 @@ const useSiteMenuItems = () => {
 	const menuItems = useSelector( ( state ) => getAdminMenu( state, selectedSiteId ) );
 
 	useEffect( () => {
-		if ( selectedSiteId !== null ) {
+		if ( selectedSiteId && siteDomain ) {
```
Comment (getdave) [inline client/my-sites/sidebar-unified/use-site-menu-items.js:28]: Nice work simplifying this.

**Review by getdave — APPROVED** [review]
Looks good. Originally functionality still preserved for fallback and new changes for all sites also work.

Nice work.

**Pull request merged**